src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/compute/v5_4/OpenNebulaComputePluginTest.java

```diff
@@ -97,761 +97,441 @@
-		HardwareRequirements flavor = createHardwareRequirements();
-		Mockito.doReturn(flavor).when(this.plugin).findSmallestFlavor(client, computeOrder);
+		Mockito.doReturn(new HashMap<String, String>()).when(this.plugin).getImagesSizes(Mockito.any(Client.class));
+		Mockito.doReturn(DISK_VALUE_30GB).when(this.plugin).getDiskSizeFromImageSizeMap(Mockito.any(Map.class), Mockito.anyString());
+		Mockito.doReturn(true).when(this.plugin).containsFlavor(Mockito.any(HardwareRequirements.class));
+		PowerMockito.when(OpenNebulaClientUtil.getTemplatePool(Mockito.any(Client.class))).thenReturn(templatePool);
 
-		OneResponse response = Mockito.mock(OneResponse.class);
-		PowerMockito.mockStatic(VirtualMachine.class);
-		PowerMockito.when(VirtualMachine.allocate(Mockito.any(Client.class), Mockito.anyString())).thenReturn(response);
-		Mockito.when(response.isError()).thenReturn(false);
-		
-		int choice = 1;
-		String networkId = null;
-		String valueOfCpu = String.valueOf(CPU_VALUE_4);
-		String valueOfRam = String.valueOf(MEMORY_VALUE_2048);
-		String valueOfDisk = String.valueOf(DISK_VALUE_8GB);
-		String template = generateTemplate(choice, valueOfCpu, valueOfRam, networkId, valueOfDisk);
+		Mockito.when(templatePool.iterator()).thenReturn(templateIterator);
+		Mockito.when(templateIterator.hasNext()).thenReturn(true, false);
+		Mockito.when(templateIterator.next()).thenReturn(template);
+		Mockito.when(template.getId()).thenReturn(flavorId);
+		Mockito.when(template.getName()).thenReturn(this.hardwareRequirements.getName());
+		Mockito.when(template.xpath(TEMPLATE_CPU_PATH)).thenReturn(String.valueOf(this.hardwareRequirements.getCpu()));
+		Mockito.when(template.xpath(TEMPLATE_MEMORY_PATH)).thenReturn(String.valueOf(this.hardwareRequirements.getMemory()));
+		Mockito.when(template.xpath(TEMPLATE_DISK_SIZE_PATH)).thenReturn("0");
+		Mockito.when(template.xpath(TEMPLATE_IMAGE_ID_PATH)).thenReturn(flavorId);
 
 		// exercise
-		this.plugin.requestInstance(computeOrder, cloudUser);
+		this.plugin.updateHardwareRequirements(this.client);
 
 		// verify
-		PowerMockito.verifyStatic(OpenNebulaClientUtil.class, VerificationModeFactory.times(1));
-		OpenNebulaClientUtil.createClient(Mockito.anyString(), Mockito.anyString());
+		PowerMockito.verifyStatic(OpenNebulaClientUtil.class, Mockito.times(TestUtils.RUN_ONCE));
+		OpenNebulaClientUtil.getTemplatePool(Mockito.eq(this.client));
 
-		Mockito.verify(this.plugin, Mockito.times(1)).findSmallestFlavor(Mockito.any(Client.class),
-				Mockito.any(ComputeOrder.class));
+		Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_ONCE)).getImagesSizes(Mockito.eq(this.client));
+		Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_THRICE)).convertToInteger(Mockito.anyString());
+		Mockito.verify(template, Mockito.times(TestUtils.RUN_ONCE)).getId();
+		Mockito.verify(template, Mockito.times(TestUtils.RUN_ONCE)).getName();
+		Mockito.verify(template, Mockito.times(4)).xpath(Mockito.anyString());
```

Comment:
Here too... set this number into a constant.

Reply:
done